src/core/client/admin/routes/Moderate/Queue/ModerationQueue.tsx

```diff
@@ -3,17 +3,36 @@ import { FunctionComponent, useEffect } from "react";
 
 interface Props {
   mode: "PRE" | "POST" | "%future added value" | null;
+  siteID?: string;
+  storyID?: string;
 }
 
-const ModerationQueue: FunctionComponent<Props> = ({ mode }) => {
+const ModerationQueue: FunctionComponent<Props> = ({
+  mode,
+  siteID,
+  storyID,
+}) => {
   const { router } = useRouter();
   useEffect(() => {
+    let path = null;
+    let suffix = "";
+
     if (mode === "PRE") {
-      router.replace("/admin/moderate/pending");
+      path = "pending";
     } else if (mode === "POST") {
-      router.replace("/admin/moderate/reported");
+      path = "reported";
+    }
+
+    if (siteID) {
+      suffix = `sites/${siteID}`;
+    } else if (storyID) {
+      suffix = `stories/${storyID}`;
+    }
+
+    if (path) {
+      router.replace(`/admin/moderate/${path}/${suffix}`);
     }
```

Comment:
You can use the helper `getModerationLink` to generate this! Would simplify this instead to:

```ts
router.replace(getModerationLink({ queue: mode === "PRE" ? "pending" : "reported", storyID, siteID }));
```